routers/repo/view.go

```diff
@@ -56,18 +56,28 @@ func renderDirectory(ctx *context.Context, treeLink string) {
 		return
 	}
 
-	var readmeFile *git.Blob
+	var readmeFiles [4]*git.Blob
```

Comment:
How can you be sure this will always be 4?

Reply:
This is for all the readme files found, the first 3 will hold files matching `{".md", ".txt", ""}` in order, the last one is for any non-strictly matching readme file. The first non-nil file in the array will be taken. This makes the priority `readme.md > readme.txt > readme > readme.<i18n>.md > ...`